### PR TITLE
feat(dashboards): Save widget type to split decision if explicit

### PR DESCRIPTION
**Pull request by narsaynorath**
If the widget type is explicitly one of our split types (i.e. error-events, transaction-like) then save that split type to the `discover_widget_split` field as well for compatibility with on-demand.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_dashboards.py [hunk 1]
```diff
@@ -861,3 +861,76 @@ def test_post_dashboard_with_widget_filter_requiring_environment(self):
         }
         response = self.do_request("post", f"{self.url}?environment=mock_env", data=data)
         assert response.status_code == 201, response.data
+
+    def test_post_dashboard_with_widget_split_datasets(self):
```
Comment (shruthilayaj) [inline tests/sentry/api/endpoints/test_organization_dashboards.py:865]: should we write a test for `put` as well?

**Review by shruthilayaj — APPROVED** [review]

File: tests/sentry/api/endpoints/test_organization_dashboards.py (hunk 1 above)
Comment (narsaynorath) [reply tests/sentry/api/endpoints/test_organization_dashboards.py:865]: The list endpoint only supports `GET` and `POST`! The `PUT` test is on the detail endpoint where you're working with a specific dashboard instance (in `tests/sentry/api/endpoints/test_organization_dashboard_details.py`), it's basically the same test

**Comment by narsaynorath** [comment]
Going to split this up

**Comment by narsaynorath** [comment]
Closed the wrong PR 🤦

**Pull request merged**